engine/combat/combat_controller.py

```diff
@@ -144,34 +145,47 @@ def execute_combat(self: Self, delta: float) -> bool:
         # Check if we have control
         return False
 
-    def _handle_starting_zone(self: Self) -> None:
-        if (
-            not self.action
-            and level_manager.current_level == "72e9f2699f7c8394b93afa1d273ce67a"
-        ):
+    def _handle_encounter_controller(self: Self) -> None:
+        if not self.action:
             for player in combat_manager.players:
                 if combat_manager.selected_character == player.character:
-                    if self.second_encounter is True:
-                        match player.character:
-                            case PlayerPartyCharacter.Valere:
-                                self.action = Action(
-                                    SoSConsideration(player),
-                                    CrescentArc(timing_type=SoSTimingType.NONE),
-                                )
-                            case PlayerPartyCharacter.Zale:
-                                self.action = Action(
-                                    SoSConsideration(player),
-                                    Sunball(value=1000, hold_time=2.0),
-                                )
-                    else:
-                        match player.character:
-                            case PlayerPartyCharacter.Valere:
-                                self.action = Action(
-                                    SoSConsideration(player),
-                                    BasicAttack(timing_type=SoSTimingType.NONE),
-                                )
-                            case PlayerPartyCharacter.Zale:
-                                self.action = Action(
-                                    SoSConsideration(player),
-                                    BasicAttack(timing_type=SoSTimingType.NONE),
-                                )
+                    match combat_manager.combat_controller:
```

Comment:
Likely we want to split this code up a bit more later, maybe into separate classes instead. Yes, there aren't that many special fights like this, but it will still add some clutter to this file.

Reply:
i agree - that will come with the combat_controller refactor when i handle custom fights or situations like this.